src/transformers/models/iris/modeling_iris.py

```diff
@@ -0,0 +1,1964 @@
+            The actions from real environment in L sequence length(timesteps)
+        rewards (`torch.FloatTensor` of shape `(batch_size, sequence_length)`):
+            The rewards from real environment in L sequence length(timesteps)
+        ends (`torch.IntTensor` of shape `(batch_size, sequence_length)`):
+            The episode termination booleans from real environment in L sequence length(timesteps)
+        mask_padding (`torch.BoolTensor` of shape `(batch_size, 1)`):
+            The padding mask for input observations before putting them in encode and decoder
+        should_preprocess (`bool`, *optional*):
+            If set to `True`, `observations` are preprocessed before being passed to the model when encoding.
+        should_postprocess(`bool`, *optional*):
+            If set to `True`, `reconstructions` are postprocessed after being passed to the model when decoding.
+        output_attentions (`bool`, *optional*):
+            Whether or not to return the attentions tensors of all attention layers. See `attentions` under returned
+            tensors for more detail.
+        output_hidden_states (`bool`, *optional*):
+            Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors for
+            more detail.
+        return_dict (`bool`, *optional*):
+            Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+
+
+"""
+
+
+class LossWithIntermediateLosses:
+    def __init__(self, **kwargs):
+        self.loss_total = sum(kwargs.values())
+        self.intermediate_losses = {k: v.item() for k, v in kwargs.items()}
+
+    def __truediv__(self, value):
+        for k, v in self.intermediate_losses.items():
+            self.intermediate_losses[k] = v / value
+        self.loss_total = self.loss_total / value
+        return self
+
+
+@dataclass
+class TokenizerEncoderOutput:
+    z: torch.FloatTensor
+    z_quantized: torch.FloatTensor
+    tokens: torch.LongTensor
+
+
+class Slicer(nn.Module):
+    def __init__(self, max_blocks: int, block_mask: torch.Tensor) -> None:
+        super().__init__()
+        self.block_size = block_mask.size(0)
+        self.num_kept_tokens = block_mask.sum().long().item()
+        kept_indices = torch.where(block_mask)[0].repeat(max_blocks)
+        offsets = torch.arange(max_blocks).repeat_interleave(self.num_kept_tokens)
+        self.register_buffer("indices", kept_indices + block_mask.size(0) * offsets)
+
+    def compute_slice(self, num_steps: int, prev_steps: int = 0) -> torch.Tensor:
+        total_steps = num_steps + prev_steps
+        num_blocks = math.ceil(total_steps / self.block_size)
+        indices = self.indices[: num_blocks * self.num_kept_tokens]
+        return indices[torch.logical_and(prev_steps <= indices, indices < total_steps)] - prev_steps
+
+    def forward(self, *args, **kwargs):
+        raise NotImplementedError
+
+
+class Head(Slicer):
```

Comment:
We don't support subclassing like this in our modeling files - all custom layers should be a child of `nn.Module`, `PretrainedModel` or `IrisPretrainedModel`

Reply:
Oh! yeah got it, forgot to change that though saw that in the contribution docs. Refactoring that asap